pkg/controller/endpointslicemirroring/endpointset.go

```diff
@@ -0,0 +1,96 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package endpointslicemirroring
+
+import (
+	"sort"
+
+	discovery "k8s.io/api/discovery/v1beta1"
+	endpointutil "k8s.io/kubernetes/pkg/controller/util/endpoint"
+)
+
+// endpointHash is used to uniquely identify endpoints. Only including addresses
+// and hostnames as unique identifiers allows us to do more in place updates
+// should attributes such as topology, conditions, or targetRef change.
+type endpointHash string
+type endpointHashObj struct {
+	Addresses []string
```

Comment:
is Addresses sorted?

Reply:
Good question - it is sorted as part of the `hashEndpoint()` function that's called for all major operations: https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/endpointslice/endpointset.go#L36.